app/grandchallenge/cases/views.py

```diff
@@ -139,3 +142,39 @@ def get_context_data(self, **kwargs):
             }
         )
         return context
+
+
+class CSImageDetail(
+    LoginRequiredMixin, ObjectPermissionRequiredMixin, DetailView
+):
+    model = Image
+    permission_required = (
+        f"{Image._meta.app_label}.view_{Image._meta.model_name}"
+    )
+    raise_exception = True
+    login_url = reverse_lazy("account_login")
+    template_name = "cases/image_detail_cs.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        try:
+            self.object.get_metaimage_files()
+        except FileNotFoundError as e:
+            raise Http404 from e
+
+        if (
+            self.object.depth and self.object.depth > 1
+        ) or self.object.color_space == Image.COLOR_SPACE_YCBCR:
+            # >2D volumes and YCrBr not supported in cornerstone
+            raise Http404
+
+        context.update({"image_pk": self.object.pk})
+        return context
+
+
+class CSImageLoader(RetrieveAPIView):
+    queryset = Image.objects.all()
+    permission_classes = (DjangoObjectPermissions,)
```

Comment:
I think that it is best that we limit access to this endpoint given how heavyweight it is. Is it possible to add a global permission to use this feature alongside the object-based permissions?